### PR TITLE
Hide some sensors from unsupported devices

### PR DESCRIPTION
**Pull request by dshokouhi**
## Summary

Hide some sensors that simply won't work in Automotive. Also hide battery manager if the device has no battery.

As these devices do not support the sensor they also wouldn't support for any registered intent so should be safe to continue hiding.

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#

### CODE REVIEW

File: app/src/main/java/io/homeassistant/companion/android/sensors/NotificationSensorManager.kt [hunk 1]
```diff
@@ -70,7 +71,11 @@ class NotificationSensorManager : NotificationListenerService(), SensorManager {
     }
     override fun hasSensor(context: Context): Boolean {
         val uiManager = context.getSystemService<UiModeManager>()
```
Comment (jpelgrom) [inline app/src/main/java/io/homeassistant/companion/android/sensors/NotificationSensorManager.kt:73]: You can move this down into the `if`, we don't need the UI manager if this is Automotive

**Review by jpelgrom — COMMENTED** [review]
I don't know, always returning `in_vehicle` for the activity sensor when using Automotive also would be an option ;)

File: app/src/main/java/io/homeassistant/companion/android/sensors/NotificationSensorManager.kt (hunk 1 above)
Comment (dshokouhi) [reply app/src/main/java/io/homeassistant/companion/android/sensors/NotificationSensorManager.kt]: thanks :)

**Review by jpelgrom — APPROVED** [review]
Good to improve the Automotive experience

**Review by JBassett — APPROVED** [review]

**Pull request merged**